packages/components/src/components/combobox/combobox.component.ts

```diff
@@ -0,0 +1,967 @@
+    // Close when focusing out of the combobox
+    const path = event.composedPath();
+    if (this && !path.includes(this)) {
+      // eslint-disable-next-line @typescript-eslint/no-floating-promises
+      this.hide();
+    }
+  };
+
+  /* eslint-disable @typescript-eslint/no-floating-promises */
+  // eslint-disable-next-line complexity
+  private handleDocumentKeyDown = (event: KeyboardEvent) => {
+    const target = event.target as HTMLElement;
+    const isClearButton = target.closest('.combobox__clear') !== null;
+
+    if (isClearButton) {
+      return;
+    }
+
+    // Close when pressing escape and open / clear input if not open
+    if (event.key === 'Escape') {
+      if (this.open && !this.closeWatcher) {
+        event.preventDefault();
+        event.stopPropagation();
+        this.hide();
+        this.displayInput.focus({ preventScroll: true });
+      } else if (!this.open) {
+        this.clearCombobox();
+      }
+    }
+
+    // Handle enter.
+    if (event.key === 'Enter') {
+      event.preventDefault();
+      event.stopImmediatePropagation();
+
+      // If it's not open, open it
+      if (!this.open) {
+        this.show();
+        return;
+      }
+
+      // If it is open, update the value based on the current selection and close it
+      if (this.currentOption && !this.currentOption.disabled) {
+        const oldValue = this.lastOptionValue;
+        this.setSelectedOption(this.currentOption);
+
+        if (this.value !== oldValue) {
+          // Emit after updating
+          this.updateComplete.then(() => {
+            this.emit('syn-input');
+            this.emit('syn-change');
+          });
+        }
+
+        this.hide();
+        this.displayInput.focus({ preventScroll: true });
+      }
+
+      return;
+    }
+
+    // Navigate options
+    if (['ArrowUp', 'ArrowDown'].includes(event.key)) {
+      // Prevent scrolling
+      event.preventDefault();
+      // Open it
+      if (!this.open) {
+        this.show();
+      }
+
+      this.selectNextOption(event.key === 'ArrowDown');
+    }
+
+    // Move cursor
+    if (['Home', 'End'].includes(event.key)) {
+      // Prevent scrolling
+      event.preventDefault();
+      if (event.key === 'Home') {
+        this.displayInput.setSelectionRange(0, 0);
+      } else if (event.key === 'End') {
+        this.displayInput.setSelectionRange(
+          this.displayInput.value.length,
+          this.displayInput.value.length,
+        );
+      }
+    }
+  };
+  /* eslint-enable @typescript-eslint/no-floating-promises */
+
+  private handleDocumentMouseDown = (event: MouseEvent) => {
+    // Close when clicking outside of the combobox
+    const path = event.composedPath();
+    if (this && !path.includes(this)) {
+      // eslint-disable-next-line @typescript-eslint/no-floating-promises
+      this.hide();
+    }
+  };
+
+  private handleLabelClick() {
+    this.displayInput.focus();
+  }
+
+  private handleComboboxMouseDown() {
```

Comment:
I don´t know if I am the only one that has a usability issue here (may be personal preference), but I would suggest to have a slight delay on this. Reason: Some people (including me ;)) use a double click mouse gesture to select the whole input text so they can remove it. When using such a gesture, it will open, close and finally reopen the dropdown.

Reply:
nice catch! Never tried this until now

Reply:
after discussion we decided to not fix the open / close of the listbox on text selection.
This would lead to other implications like clicking on the combobox will open the listbox delayed, which feels very strange.
Also others also don`t do anything here like spectrum, whose component has the same problem